src/services/TestResultsService.ts

```diff
@@ -62,16 +63,22 @@ import {
   testTypesSchemaSpecTestGroup1, testTypesSchemaSpecTestGroup2,
   testTypesSchemaSpecTestGroup3, testTypesSchemaSpecTestGroup4, testTypesSchemaSpecTestGroup5
 } from "../models/test-types/testTypesSchemaSpecialistTestsPut";
+import { Injector } from "../models/injector/Injector";
+import { TestDataProvider } from "../handlers/expiry/providers/TestDataProvider";
+import { TestTypeForExpiry } from "../models/TestTypeforExpiry";
+import { DateProvider } from "../handlers/expiry/providers/DateProvider";
 
 /**
  * Service for retrieving and creating Test Results from/into the db
  * @returns Promise
  */
 export class TestResultsService {
   public readonly testResultsDAO: TestResultsDAO;
-
+  public vehicleTestController: VehicleTestController;
   constructor(testResultsDAO: TestResultsDAO) {
     this.testResultsDAO = testResultsDAO;
+    this.vehicleTestController = Injector.resolve<VehicleTestController>(VehicleTestController, [TestDataProvider, DateProvider]);
+    this.vehicleTestController.dataProvider.testResultsDAO = this.testResultsDAO;
```

Comment:
I think we flip the !filters to throw we might get away with one nesting condition if it worked

Reply:
If we use optional chaining we could also remove more nesting. Could be out of scope so happy to discuss and get your take on that please
`filters?.fromDateTime` etc..